File: app/src/main/java/com/example/dagger_hilt/data/datasource/web/api/MovieService.java
package com.example.dagger_hilt.data.datasource.web.api;

import androidx.lifecycle.LiveData;

import com.example.dagger_hilt.data.datasource.web.models.response.ApiResponse;

import retrofit2.http.GET;
import retrofit2.http.Query;

public interface MovieService {
    @GET("discover/movie")
    LiveData<ApiResponse> loadMovies(@Query("api_key") String apiKey);

}

File: app/src/main/java/com/example/dagger_hilt/data/datasource/web/api/MovieService.kt
package com.example.dagger_hilt.data.datasource.web.api

import com.example.dagger_hilt.data.datasource.web.models.MoviesResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface MovieService {
    @GET("discover/movie")
    suspend fun loadMovies(
        @Query("api_key") apiKey: String
    ): Response<MoviesResponse>
}